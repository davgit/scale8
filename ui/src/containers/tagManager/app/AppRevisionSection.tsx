import { FC } from 'react';
import { useQuery } from '@apollo/client';
import TagIcon from '../../../components/atoms/Icons/TagIcon';
import PlatformRevisionIcon from '../../../components/atoms/Icons/PlatformRevisionIcon';
import NavAppRevisionQuery from '../../../gql/queries/NavAppRevisionQuery';
import { NavAppRevision } from '../../../gql/generated/NavAppRevision';
import {
    buildAppButtonProps,
    buildAppRevisionButtonProps,
    buildOrgButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import ActionIcon from '../../../components/atoms/Icons/ActionIcon';
import TriggerIcon from '../../../components/atoms/Icons/TriggerIcon';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { toAppRevision } from '../../../utils/NavigationPaths';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';

const AppRevisionSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();
    const { ask, dispatchDialogAction, setRefreshCurrentPage } = templateInteractions;

    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavAppRevision> = {
        children,
        sectionKey: SectionKey.appRevision,
        queryResult: useQuery<NavAppRevision>(NavAppRevisionQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(analyticsEnabled(data.getRevision.app));
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getRevision.app.tag_manager_account.org.id,
                data.getRevision.app.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getRevision.app.tag_manager_account.id,
                data.getRevision.app.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getRevision.app.tag_manager_account.apps,
                data.getRevision.app.id,
                data.getRevision.app.name,
            ),
            buildAppRevisionButtonProps(
                router,
                data.getRevision.app.revisions,
                id,
                data.getRevision.name,
                true,
            ),
        ],
        buildMenuItemsProps: () => [
            {
                icon: () => <TagIcon />,
                label: 'Tags',
                link: toAppRevision({ id }, 'tags'),
            },
            {
                icon: () => <TriggerIcon />,
                label: 'Global Triggers',
                link: toAppRevision({ id }, 'global-triggers'),
            },
            {
                icon: () => <ActionIcon />,
                label: 'Global Actions',
                link: toAppRevision({ id }, 'global-actions'),
            },
            {
                icon: () => <PlatformRevisionIcon />,
                label: 'Platform Revisions',
                link: toAppRevision({ id }, 'app-platform-revisions'),
            },
        ],
        extractOrgUserDetails: (data) => data.getRevision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                id,
                data.getRevision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getRevision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavAppRevision> {...sectionProps} />;
};

export default AppRevisionSection;
