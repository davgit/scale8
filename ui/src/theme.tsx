import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
