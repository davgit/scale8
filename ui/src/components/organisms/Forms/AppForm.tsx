import { FC, useEffect, useState } from 'react';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import { Box, Checkbox, Divider, FormControlLabel, useTheme } from '@material-ui/core';
import Loader from '../Loader';
import { AppFormProps } from '../../../dialogPages/tagManager/app/AppCreate';
import StorageProviderSelector from '../../molecules/StorageProviderSelector';
import CheckBoxInput from '../../atoms/InputTypes/CheckBoxInput';
import { Mode } from '../../../gql/generated/globalTypes';
import { useConfigState } from '../../../context/AppContext';

const AppForm: FC<AppFormProps> = (props: AppFormProps) => {
    const { mode } = useConfigState();
    const theme = useTheme();

    const [useDomainName, setUseDomainName] = useState(
        props.useDomainName === undefined ? true : props.useDomainName,
    );

    const { values, handleChange } = props;

    useEffect(() => {
        if (values.domain !== '' && useDomainName && values.name !== props.values.domain) {
            handleChange('name', values.domain);
        }
    }, [values, handleChange, useDomainName]);

    return (
        <>
            {props.isSubmitting && props.isCreate ? (
                <Box display="flex" flexDirection="column" height="100vh">
                    <Box display="flex" alignItems="center" height={44}>
                        <Box ml={3} flexGrow={1} fontSize="h6.fontSize" display="flex">
                            Creating your application
                        </Box>
                    </Box>
                    <Divider />
                    <Box flex={1} position="relative" width="100%">
                        <Box p={3} height="100%" width="100%" position="absolute" overflow="auto">
                            <p>
                                Please wait while we build the data structure of your new
                                application.
                            </p>
                            <p>This typically only takes a few seconds.</p>
                            <Loader />
                        </Box>
                    </Box>
                </Box>
            ) : (
                <DrawerFormLayout {...props}>
                    <ControlledTextInput
                        name="domain"
                        label="Domain"
                        formProps={props}
                        className="DrawerFormField"
                        required
                    />
                    <FormControlLabel
                        style={{ marginBottom: theme.spacing(3) }}
                        control={
                            <Checkbox
                                name="useDomainName"
                                checked={useDomainName}
                                onChange={(event) => {
                                    setUseDomainName(event.target.checked);
                                }}
                                color="primary"
                            />
                        }
                        label="Use domain as Application name"
                    />
                    {!useDomainName && (
                        <ControlledTextInput
                            name="name"
                            label="Name"
                            formProps={props}
                            className="DrawerFormField"
                            required
                        />
                    )}

                    {mode !== Mode.COMMERCIAL && (
                        <>
                            <CheckBoxInput
                                name="analyticsEnabled"
                                value={props.values.analyticsEnabled}
                                setValue={(v) => {
                                    props.handleChange('analyticsEnabled', v);
                                }}
                                label="Enable Analytics"
                                className="DrawerFormField"
                                style={{ marginLeft: '-11px' }}
                                color="primary"
                            />
                            <small className="DrawerFormField" style={{ marginTop: '-24px' }}>
                                Enable simple analytics dashboard when using Tag Manager. We
                                recommend this option is left enabled.
                            </small>
                            <CheckBoxInput
                                name="errorTrackingEnabled"
                                value={props.values.errorTrackingEnabled}
                                setValue={(v) => {
                                    props.handleChange('errorTrackingEnabled', v);
                                }}
                                label="Enable Error Tracking"
                                className="DrawerFormField"
                                style={{ marginLeft: '-11px' }}
                                color="primary"
                            />
                            <small className="DrawerFormField" style={{ marginTop: '-24px' }}>
                                Enable basic website and mobile website application errors directly
                                in the dashboard. If you are using another service you may wish to
                                disable this option.
                            </small>
                            {(props.values.analyticsEnabled ||
                                props.values.errorTrackingEnabled) && (
                                <StorageProviderSelector
                                    {...props}
                                    warnGraphDisabled
                                    infoText="Your analytics and error data will be sent to your
                                            chosen storage provider. MongoDB is a great solution for
                                            very low traffic websites and doesn't require any
                                            configuration unless you wish to specify your own
                                            MongoDB servers. We strongly recommend using BigQuery or
                                            Clickhouse for larger websites."
                                />
                            )}
                        </>
                    )}
                </DrawerFormLayout>
            )}
        </>
    );
};

export default AppForm;
