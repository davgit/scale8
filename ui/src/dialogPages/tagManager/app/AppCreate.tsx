import { FC } from 'react';
import AppForm from '../../../components/organisms/Forms/AppForm';
import { FormProps, FormValidationResult } from '../../../hooks/form/useFormValidation';
import { ApolloError, useMutation } from '@apollo/client';
import CreateAppQuery from '../../../gql/mutations/CreateAppQuery';
import { AppType } from '../../../gql/generated/globalTypes';
import domainValidator from '../../../utils/validators/domainValidator';
import nameValidator from '../../../utils/validators/nameValidator';
import { CreateApp } from '../../../gql/generated/CreateApp';
import { DialogPageProps } from '../../../types/DialogTypes';
import { buildStandardFormInfo } from '../../../utils/InfoLabelsUtils';
import { DialogForm, DialogFormProps } from '../../abstractions/DialogForm';

export type AppValues = {
    name: string;
    domain: string;
    type: AppType;
    analyticsEnabled?: boolean;
    errorTrackingEnabled?: boolean;
    storageProvider?: string;
    bucketName?: string;
    accessKeyId?: string;
    secretAccessKey?: string;
    region?: string;
    pathPrefix?: string;
    serviceAccountJSON?: string;
    dataSetName?: string;
    requirePartitionFilterInQueries?: boolean;
    connectionString?: string;
    databaseName?: string;
};

export type AppFormProps = FormProps<AppValues> & {
    useDomainName?: boolean;
    isCreate: boolean;
};

const AppCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    const accountId = props.contextId;

    const appCreateProps: DialogFormProps<AppValues, AppFormProps, CreateApp> = {
        buildInitialState: () => ({
            name: '',
            domain: '',
            type: AppType.WEB,
        }),
        saveQuery: useMutation(CreateAppQuery),
        mapSaveData: (appValues: AppValues) => ({
            appCreateInput: {
                tag_manager_account_id: accountId,
                name: appValues.name,
                domain: appValues.domain,
                type: appValues.type,
            },
        }),
        buildFormProps: (
            formValidationValues: FormValidationResult<AppValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Create Application',
            title: 'Create Application',
            formInfoProps: buildStandardFormInfo('applications', 'Create'),
            handleDialogClose: props.handleDialogClose,
            isCreate: true,
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.createApp.id !== undefined,
        getSavedId: (formMutationData) => formMutationData?.createApp.id,
        pageComponent: AppForm,
        validators: [
            {
                field: 'name',
                validator: nameValidator,
                error: () => 'Application name too short',
            },
            {
                field: 'domain',
                validator: domainValidator,
                error: () => 'Invalid domain',
            },
        ],
        ...props,
    };

    return <DialogForm<AppValues, AppFormProps, CreateApp> {...appCreateProps} />;
};

export { AppCreate };
