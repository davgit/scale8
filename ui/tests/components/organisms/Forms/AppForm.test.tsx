import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import AppForm from '../../../../src/components/organisms/Forms/AppForm';
import { AppValues } from '../../../../src/dialogPages/tagManager/app/AppCreate';
import { AppType } from '../../../../src/gql/generated/globalTypes';

const MockForm: FC = () => {
    const formProps = useMockFormProps<AppValues>({
        name: '',
        domain: '',
        type: AppType.WEB,
        analyticsEnabled: true,
        errorTrackingEnabled: true,
        storageProvider: 'AWS_KINESIS',
        region: '',
    });

    return <AppForm {...formProps} isCreate={true} />;
};

describe('AppForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
