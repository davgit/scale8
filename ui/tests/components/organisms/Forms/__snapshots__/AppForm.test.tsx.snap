// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Domain
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                name="domain"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-18yztu3-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="useDomainName"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
            >
              Use domain as Application name
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd DialogFormField css-fbaq5x-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="analyticsEnabled"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
            >
              Enable Analytics
            </span>
          </label>
          <small
            class="DialogFormField MuiBox-root css-mnf576"
          >
            Enable simple analytics dashboard when using Tag Manager. We recommend this option is left enabled.
          </small>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd DialogFormField css-fbaq5x-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="errorTrackingEnabled"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
            >
              Enable Error Tracking
            </span>
          </label>
          <small
            class="DialogFormField MuiBox-root css-mnf576"
          >
            Enable basic website and mobile website application errors directly in the dashboard. If you are using another service you may wish to disable this option.
          </small>
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Storage Provider
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-storageProvider"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-storageProvider"
                role="button"
                tabindex="0"
              >
                Amazon Kinesis
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="storageProvider"
                required=""
                tabindex="-1"
                value="AWS_KINESIS"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard DialogFormField css-1sd9cqf-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-17ii7u1-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-acap47-MuiAlert-message"
            >
              Your analytics and error data will be sent to your chosen storage provider. MongoDB is a great solution for very low traffic websites and doesn't require any configuration unless you wish to specify your own MongoDB servers. We strongly recommend using BigQuery or Clickhouse for larger websites.
            </div>
          </div>
          <div
            class="MuiBox-root css-17ck9e8"
          >
            <div
              class="MuiBox-root css-79elbk"
            >
              <div
                class="MuiBox-root css-rxw2ex"
              >
                AWS Kinesis Storage Config
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Stream Name
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="streamName"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Access Key Id
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="accessKeyId"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Secret Access Key
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="secretAccessKey"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Region
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-1ejkwvn-MuiInputBase-root-MuiInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-component-select-region"
                  class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  id="mui-component-select-region"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="region"
                  required=""
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
